c09c2b3a42227c3fe60ca9f8ca6fe5ee7c72a69a
src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java
--- 
+++ 
@@ -233,12 +233,6 @@
   }
 
   @Override
-  public void setStateDir(Path source) throws IOException {
-    // only called by legacy restore
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
   public Set<String> getIndexNames() {
     return immutableState.globalStateInfo.getIndicesMap().keySet();
   }
@@ -448,7 +442,6 @@
             getConfiguration().getArchiveDirectory(),
             getConfiguration().getBackupWithInArchiver(),
             getConfiguration().getRestoreFromIncArchiver(),
-            getConfiguration().getStateConfig().useLegacyStateManagement(),
             indexStateManager,
             getConfiguration().getDiscoveryFileUpdateIntervalMs());
     try {
@@ -490,11 +483,6 @@
     indexStateManager.close();
 
     return DummyResponse.newBuilder().setOk("ok").build();
-  }
-
-  @Override
-  public synchronized void indexClosed(String name) {
-    throw new UnsupportedOperationException();
   }
 
   @Override